Application/src/controller/network/Decode.java

```diff
@@ -32,6 +32,49 @@ void execute() {
         // If no string has been allocated, abort.
         if (rawStringToDecode == null) return;
 
+        // Deployment diagram has been found.
+        if (rawStringToDecode.contains("deployment_diagram")) {
+
+            // Remove the curly brackets.
+            rawStringToDecode = removeCharactersFromString(rawStringToDecode, '{', '}', '"');
+
+            // Removes whitespaces and new lines.
+            rawStringToDecode = rawStringToDecode.replaceAll("\n", "").replaceAll("\\s", "");
+
+            // Splits the input.
+            String[] split = rawStringToDecode.split(",");
+
+            // Start on the third value (ignoring the first, second and third.)
+            for (int i = 3; i < split.length; i++) {
+
+                // Remove not required characters.
+                String field = removeCharactersFromString(split[i], '[', ']');
+                String second = "";
+
+                if (!field.contains("]")) {
```

Comment:
what does this check for?

Reply:
The input can be either:
['device'] or ['device','process']
Because I split with ','
It becomes ["['device'", "'process']"]
The first one always has [, But depending on the number of items it may not always have a ].
Basically looking if it has a pair or not

Reply:
I meant comment on the code bc it wasn't immediately clear